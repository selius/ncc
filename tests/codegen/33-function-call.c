int printf(const int *fmt);

void a()
{
	printf("a() called\n");
}

void b(int a)
{
	printf("b() called\n");
	__print_int(a);
}

int c()
{
	printf("c() called\n");
	return 10;
}

float d(float a)
{
	printf("d() called\n");
	return a * 5;
}

int main()
{
	a();

	b(42);

	__print_int(c());
	__print_int(c() + 5);

	__print_float(d(0.5));

	__print_float(d(c()));
	
	return 0;
}
