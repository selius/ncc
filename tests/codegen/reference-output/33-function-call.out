a() called
b() called
42
c() called
10
c() called
15
d() called
2.500000
c() called
d() called
50.000000
